sources/packages/backend/apps/api/src/route-controllers/student-account-applications/_test_/e2e/student-account-application.aest.controller.approveStudentAccountApplication.e2e-spec.ts

```diff
@@ -0,0 +1,155 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../../testHelpers";
+import {
+  E2EDataSources,
+  createE2EDataSources,
+  createFakeStudentAccountApplication,
+  createFakeUser,
+  getProviderInstanceForModule,
+  saveFakeSFASIndividual,
+} from "@sims/test-utils";
+import { AppAESTModule } from "../../../../app.aest.module";
+import { FormNames, FormService } from "../../../../services";
+import { NotificationMessageType } from "@sims/sims-db";
+
+describe("StudentAccountApplicationAESTController(e2e)-approveStudentAccountApplication", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+  let sharedFormService: FormService;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource, module } =
+      await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+
+    sharedFormService = await getProviderInstanceForModule(
+      module,
+      AppAESTModule,
+      FormService,
+    );
+
+    // Insert a fake email contact to send ministry email.
+    await db.notificationMessage.update(
+      { id: NotificationMessageType.PartialStudentMatchNotification },
+      { emailContacts: ["dummy@some.domain"] },
+    );
+    await db.notification.clear();
```

Comment:
The API tests so far are executed in parallel and we should not interfere with potential data used by other tests.
Truncating the notification should impact other test data that would rely on the data.
One alternative would be to put the notification-type records tested in a state that would not interfere with the assertions.
For instance, the test can try to find the newly saved notification, which would have the date sent as `null`.
The data "reset" before every test can be as below (it can be added to `beforeEach`).
```ts
// Force the notification to be marked as sent to not conflict with the tests.
await db.notification.update(
  {
    dateSent: IsNull(),
    notificationMessage: {
      id: NotificationMessageType.PartialStudentMatchNotification,
    },
  },
  {
    dateSent: null,
  },
);
```

Reply:
I set the other null dates to now, and then fetched on the null to differentiate the new notification.